core/lib/resources/presence/sentry.js

```diff
@@ -1,159 +1,270 @@
-var logging = require('minilog')('radar:presence:sentry'),
-    Persistence = require('persistence');
+var _ = require('underscore'),
+    Minilog = require('minilog'),
+    logging = Minilog('radar:sentry'),
+    Persistence = require('persistence'),
+    redisSentriesKey = 'sentry:/radar';
 
-function parse(message) {
-  try {
-    return JSON.parse(message);
+var defaultOptions = {
+  DEFAULT_EXPIRY_OFFSET: 60000,
+  REFRESH_INTERVAL: 10000,
+  CHECK_INTERVAL: 30000
+};
+
+var parseJSON = function(message) {
+  try { 
+    return JSON.parse(message); 
   } catch (e) { }
-}
+};
+
+var messageIsExpired = function(message) {
+  return (!message || !message.expiration || (message.expiration <= Date.now()));
+};
 
-function Sentry(name) {
-  this.name = name;
+var messageExpiration = function(message) {
+  return (message && message.expiration && (message.expiration - Date.now()));
+};
+
+var Sentry = function(name, options) {
   this.sentries = {};
-}
+  this._setName(name);
 
-Sentry.expiry = 20000;
-Sentry.channel = 'sentry:/radar';
+  this._defaultExpiryOffset = defaultOptions.DEFAULT_EXPIRY_OFFSET;
+  this._refreshInterval = defaultOptions.REFRESH_INTERVAL;
+  this._checkInterval = defaultOptions.CHECK_INTERVAL;
+
+  this._applyOptions(options);
+};
 
 require('util').inherits(Sentry, require('events').EventEmitter);
 
-Sentry.prototype._listen = function() {
-  var self = this;
-  if (!this.listener) {
-    this.listener = function(channel, m) {
-      if (channel != Sentry.channel) {
-        return;
-      }
-      self._save(parse(m));
-    };
-    Persistence.pubsub().on('message', this.listener);
+Sentry.prototype.start = function(options, callback) {
+  var self = this,
+      keepAliveOptions = {},
+      upMessage;
+  
+  options = options || {};
+
+  if (typeof(options) === 'function') { 
+    callback = options; 
+  } else {
+    this._applyOptions(options); 
   }
-  if (!this.subscribed) {
-    this.subscribed = true;
-    Persistence.pubsub().subscribe(Sentry.channel);
+
+  if (this._refreshTimer) { return; }
+
+  logging.info('#presence - #sentry - starting', this.name);
+
+  if (options.expiration) {
+    keepAliveOptions.expiration = options.expiration;
   }
+
+  upMessage = self._keepAlive(keepAliveOptions);
+
+  this.setMaxListeners(0);
+  this._loadAndCleanUpSentries(function() {
+    self.emit('up', this.name, upMessage);
+    Persistence.publish(redisSentriesKey, upMessage);
+    self._startListening();
+    if (callback) { callback(); }
+  });
+
+  this._refreshTimer = setTimeout(this._refresh.bind(this), Math.floor(this._refreshInterval/2));
+  this._checkSentriesTimer = setTimeout(this._checkSentries.bind(this), Math.floor(this._checkInterval/2));
```

Comment:
ok, I am amending this to include refresh as well. I see that the refresh interval is already very aggressive. So no /2 anywhere.